### PR TITLE
B2b architecture phase 1

### PR DESCRIPTION
**Pull request by tamigoodall**
Deployment link: https://docs-content-staging-pr-7798.herokuapp.com/docs/architecture-scenarios/implementation/b2b/b2b-architecture

### CODE REVIEW

File: articles/architecture-scenarios/_includes/_architecture/_introduction.md [hunk 1]
```diff
@@ -22,5 +25,9 @@ In addition, you’ll undoubtedly have an established set of processes and proce
...
 
 ::: panel Best Practice
-Other groups within your organisation may also be working with Auth0; it’s not uncommon for our customers to have disparate departments that serve different user communities. Identifying these will potentially influence your design choices, and doing so early could mitigate decisions that might prove costly later on.
+Other groups within your organization may also be working with Auth0; it’s not uncommon for our customers to have disparate departments that serve different user communities. Identifying these will potentially influence your design choices, and doing so early could mitigate decisions that might prove costly later on.
 :::
+
+<% if (platform === "b2b") { %>
+If your customers' organization supports the use of mutliple IDPs, then you should create separate Auth0 tenants for that organization; see [Tenant provision for complex organizations](#tenant-provision-for-complex-organizations) for further details. This allows you to keep the rest of your setup much simpler by maintaining a one-to-one connection relationship between your organization and all your customer organizations within your main tenant.
```
Comment (mostekcm) [inline articles/architecture-scenarios/_includes/_architecture/_introduction.md:32]:
```suggestion
If your customers' organization supports the use of mutliple IDPs, then you can create separate Auth0 tenants for that organization; see [Tenant provision for complex organizations](#tenant-provision-for-complex-organizations) for further details. This allows you to keep the rest of your setup much simpler by maintaining a one-to-one connection relationship between your organization and all your customer organizations within your main tenant. 
```

File: articles/architecture-scenarios/implementation/b2b/b2b-architecture.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+<%= include('../../_includes/_architecture/_tenant-provision.md', { platform: 'b2b' }) %>
+
+### Tenant provision for complex organizations
+
+In most cases, provisioning separate Auth0 tenants for your customer's organizations is not necessary. However, in certain circumstances this can be something that is valuable for reducing the complexity of your setup. For instance, we recommend provisioning a separate Auth0 tenant for your customers' organization as a best practice if:
+
+* Your customers' organizations have isolated users that aren't shared with other organizations.
+* You have some customer organizations that support more than one IDP. For example, your customer has their own IDP but also has some users that aren't in their IDP and whose credentials you will need to store. Or, your customer wants to provide for one or more social connections in addition to their enterprise IDP. 
+
+If both of these situations are the case, then you should create separate Auth0 tenants for each customer that needs it. This allows you to have a separate custom domain for them and to easily customize their login experience, including Home Realm Discovery on their login page.
```
Comment (mostekcm) [inline articles/architecture-scenarios/implementation/b2b/b2b-architecture.md:28]:
```suggestion
If both of these situations are the case, then you can create separate Auth0 tenants for each customer that needs it. This allows you to have a separate custom domain for them and to easily customize their login experience, including Home Realm Discovery on their login page. 
```

File: articles/architecture-scenarios/implementation/b2b/b2b-architecture.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+<%= include('../../_includes/_architecture/_custom-domains.md', { platform: 'b2b' }) %>
+
+## SDLC support
+
+<%= include('../../_includes/_architecture/_sdlc-support.md', { platform: 'b2b' }) %>
+
+## Planning
+
+To help you with planning your implementation, we've put together some [planning guidance]((https://drive.google.com/a/auth0.com/file/d/17XVLT3kU7vG49Ix9afJV0pAI9ev4oRUL/view?usp=sharing)) that details our recommended strategies.
```
Comment (mostekcm) [inline articles/architecture-scenarios/implementation/b2b/b2b-architecture.md:48]: I have a planning include now.  Will have to remember to migrate to it so we get the right URL after that gets merged.

**Review by mostekcm — CHANGES_REQUESTED** [review]
See individual comments for change requests

File: articles/architecture-scenarios/_includes/_architecture/_introduction.md (hunk 1 above)
Comment (PeterGFernandez) [inline articles/architecture-scenarios/_includes/_architecture/_introduction.md:32]: Aligned with the suggestion in PR 7801

```suggestion
If your customers' organization supports the use of mutliple IDPs, then we recommend that you create separate Auth0 tenants for that organization; see [Tenant provision for complex organizations](#tenant-provision-for-complex-organizations) for further details. This allows you to keep the rest of your setup much simpler by maintaining a one-to-one connection relationship between your organization and all your customer organizations within your main tenant. 
```

Comment (tamigoodall) [reply articles/architecture-scenarios/_includes/_architecture/_introduction.md:32]: Added to PR 7818

**Pull request merged**